homeassistant/config.py

```diff
@@ -663,6 +666,27 @@ async def async_process_ha_core_config(hass: HomeAssistant, config: dict) -> Non
     if CONF_UNIT_SYSTEM in config:
         hac.units = get_unit_system(config[CONF_UNIT_SYSTEM])
 
+    async def load_translations(_event: Event) -> None:
+        language = hass.config.language
+        _LOGGER.debug(f"Loading translations for language: {language}")
+        await async_load_state_translations_to_cache(hass, language)
```

Comment:
Should we flush the translation cache to not keep unwanted translations in memory?

Reply:
I don't think they were flushed before, also they are probably still used by the frontend as UI language is set in user's config.

But I can try to add it if you think it is beneficial.